lib/rules/jsx-fragments.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @fileoverview Enforce shorthand or standard form for React fragments.
+ * @author Alex Zherdev
+ */
+'use strict';
+
+const elementType = require('jsx-ast-utils/elementType');
+const pragmaUtil = require('../util/pragma');
+const variableUtil = require('../util/variable');
+const versionUtil = require('../util/version');
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+function replaceNode(source, node, text) {
+  return `${source.slice(0, node.range[0])}${text}${source.slice(node.range[1])}`;
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforce shorthand or standard form for React fragments',
+      category: 'Stylistic Issues',
+      recommended: false,
+      url: docsUrl('jsx-fragments')
+    },
+    fixable: 'code',
+
+    schema: [{
+      enum: ['syntax', 'element']
+    }]
+  },
+
+  create: function(context) {
+    const configuration = context.options[0] || 'syntax';
+    const sourceCode = context.getSourceCode();
+    const reactPragma = pragmaUtil.getFromContext(context);
+    const fragmentPragma = pragmaUtil.getFragmentFromContext(context);
+    const openFragShort = '<>';
+    const closeFragShort = '</>';
+    const openFragLong = `<${reactPragma}.${fragmentPragma}>`;
+    const closeFragLong = `</${reactPragma}.${fragmentPragma}>`;
+
+    function reportOnReactVersion(node) {
+      if (!versionUtil.testReactVersion(context, '16.2.0')) {
+        context.report({
+          node,
+          message: 'Fragments are only supported starting from React v16.2'
```

Comment:
Could we add something along the lines of:

```
Please disable the `jsx-fragments` rule or upgrade your version of React.
```

I think it's always a good idea to give users an action to do to help them resolve the error/warning.

Reply:
That sounds like a great improvement.